src/app.js

```diff
@@ -87,21 +91,39 @@ i18next
 
 app.use(i18nextMiddleware.handle(i18next));
 
+app.use(cookieParser());
+
+// Generate a new session ID asynchronously if no session cookie
+// `express-session` does not support async session ID generation
+// https://github.com/expressjs/session/issues/107
+app.use(async (req, res, next) => {
+  if (!req.cookies[SESSION_COOKIE_NAME]) {
+    req.generatedSessionId = await uid(24);
+  }
+  next();
+});
+
 app.use(
   session({
-    name: "ipv_core_service_session",
+    name: SESSION_COOKIE_NAME,
     store: sessionStore,
     saveUninitialized: false,
     secret: SESSION_SECRET,
     unset: "destroy",
     resave: false,
     cookie: {
-      name: "ipv_core_service_session",
+      name: SESSION_COOKIE_NAME,
       expires: false,
       secret: SESSION_SECRET,
       signed: true,
       secure: "auto",
     },
+    // Use the newly generated session ID, or fall back to the default behaviour
+    genid: (req) => {
+      const sessionId = req.generatedSessionId || uid.sync(24);
```

Comment:
When would we need to fallback?

Reply:
If the user has a session cookie but it's invalid for some reason (e.g. it's an old session, or they've just messed with the cookie themselves), then we'll still need to generate a new session.

We could try and work that all out above, but it doesn't seem worth it.